test/journalpump/senders/test_aws_cloudwatch.py

```diff
@@ -0,0 +1,120 @@
+from botocore.stub import Stubber
+from journalpump.senders import AWSCloudWatchSender
+from unittest import mock
+
+import boto3
+
+
+def test_awscloudwatch_sender():
+    logs = boto3.client("logs", region_name="us-east-1")
+
+    with Stubber(logs) as stubber:
+        stubber.add_client_error("create_log_group", service_error_code="ResourceAlreadyExistsException")
+        stubber.add_response("create_log_stream", {"ResponseMetadata": {"HTTPStatusCode": 200}})
+        stubber.add_response(
+            "describe_log_streams", {"logStreams": [{
+                "logStreamName": "stream",
+                "uploadSequenceToken": "token"
+            }]}, {"logGroupName": "group"}
+        )
+        sender = AWSCloudWatchSender(
+            name="awscloudwatch",
+            reader=mock.Mock(),
+            stats=mock.Mock(),
+            field_filter=None,
+            config={
+                "aws_cloudwatch_log_group": "group",
+                "aws_cloudwatch_log_stream": "stream"
+            },
+            aws_cloudwatch_logs=logs
+        )
+        assert sender._next_sequence_token == "token"  # pylint: disable=protected-access
+
+    with Stubber(logs) as stubber:
+        stubber.add_response(
+            "put_log_events", {
+                "ResponseMetadata": {
+                    "HTTPStatusCode": 200
+                },
+                "nextSequenceToken": "token1",
+                "rejectedLogEventsInfo": {}
+            }
+        )
+        sender.send_messages(messages=[b'{"REALTIME_TIMESTAMP": 1590581737.308352}'], cursor=None)
+        assert sender._next_sequence_token == "token1"  # pylint: disable=protected-access
+        assert sender._sent_count == 1  # pylint: disable=protected-access
```

Comment:
Ideally we don't want to test internal behaviour. A refactoring of AWSCloudWatchSender will result in a refactor of the tests I am not familiar with stubber, but I would expect these tests to check the parameters passed to the API rather than the class' internal state

Reply:
I did in fact refactor a substantial amount of all the senders before realizing that it wasn't reasonable to try and get them merged while I'm really just trying to get the typechecking running in CI for us. Really this test should be replaced with a `systest` which just calls the actual AWS Cloudwatch API if we're going to support it and then we can treat the unit tests like a blackbox with well-specified typechecks to keep them honest.